package/src/index.ts

```diff
@@ -0,0 +1,537 @@
+	/**
+	 * Number of used API requests per 60s.
+	 * @remarks
+	 * `30 req/min` on free accounts, and `60 req/min` on regular token
+	 */
+    used: number
+	/**
+	 * Remaining number of API requests per 60s.
+	 * @remarks
+	 * `30 req/min` on free accounts, and `60 req/min` on regular token
+	 */
+    remaining: number
+	/**
+	 * In how many seconds (every 60s) your rate limit is going to reset.
+	 * @remarks
+	 * `30 req/min` on free accounts, and `60 req/min` on regular token
+	 */
+    reset: number
+}
+
+/**
+ * Base API endpoint
+ */
+const baseUrl = "https://api.henrikdev.xyz/valorant"
+export default class {
+    /**
+	 * Images for every {@link Rank}
+	 * @remarks
+	 * You must call {@link initUtils} before using this!
+	 */
+    rankImages?: { [key in Rank]: { large: string, small: string, triangleUp: string, triangleDown: string } }
+    /**
+	 * Images for every {@link ValorantMap}
+	 * @remarks
+	 * You must call {@link initUtils} before using this!
+	 */
+    mapImages?: { [key in ValorantMap | "The Range"]: { splash: string, minimap: string, landscape: string } }
+
+    /**
+	 * Create a new instance of the main API. All API calls are in this class
+	 * @example
+	 * Create a new instance
+	 * ```js
+	 * import _VAPI from "unofficial-valorant-api"
+	 * const VAPI = new _VAPI("my super secret token")
+	 * ```
+	 * @param token - (optional) The token, if you have one. Get one from the Discord server ({@link https://discord.gg/wXNMnqzvAD})
+	 */
+    constructor(private token?: string) {}
+
+    /**
+	 * Creates {@link default.rankImages} and {@link default.mapImages} by pulling the images from {@link https://valorant-api.com}
+	 * @remarks
+	 * Must be called before using {@link default.rankImages} and {@link default.mapImages}
+	 */
+    async initUtils(): Promise<void> {
+        const mapsReq = (await axios({ url: "https://valorant-api.com/v1/maps" }).catch(() => null))?.data?.data
+        if (mapsReq) {
+            this.mapImages = {} as any
+            mapsReq.forEach((map: any) => (this.mapImages[map.displayName] = {
+                splash: map.splash,
+                minimap: map.displayIcon,
+                landscape: map.listViewIcon
+            }))
+        }
+
+        const tiersReq = (await axios({ url: "https://valorant-api.com/v1/competitivetiers" }).catch(() => null))?.data?.data
+        if (tiersReq) {
+            const capitalize = (string: string) => string.charAt(0).toUpperCase() + string.slice(1)
+            this.rankImages = {} as any
+            tiersReq[tiersReq.length - 1].tiers.forEach((rank: any) => (this.rankImages[capitalize(rank.tierName.toLowerCase())] = {
+                large: rank.largeIcon,
+                small: rank.smallIcon,
+                triangleUp: rank.rankTriangleUpIcon,
+                triangleDown: rank.rankTriangleDownIcon
+            }))
+        }
+    }
+
+    /**
+	 * @internal
+	 * Parses the body of an axios response
+	 */
+    private _parseBody(body: any): any {
```

Comment:
Why starting the method name with a underscore? The method is already declared as private via the `private` keyword. I know it's convention in some other languages, also Javascript. But in Typescript you don't need that underscore to indicate a private method.

Reply:
Didn't realize it was bad practice to do this :P I've personally seen and used this in TS projects, so I just used it out of habit.